bigchaindb/core.py

```diff
@@ -209,15 +209,14 @@ def end_block(self, request_end_block):
         else:
             self.block_txn_hash = block['app_hash']
 
-        # Check if the current block concluded any validator elections and
-        # update the locally tracked validator set
-        validator_update = ValidatorElection.approved_update(self.bigchaindb,
-                                                             self.new_height,
-                                                             self.block_transactions)
+        # Check the current block to see if any elections have concluded.
+        concluded_elections = Election.approved_elections(self.bigchaindb,
```

Comment:
We should directly return the validator set update from `approved_elections` since preparing such an update based on the given block is its concern.

Reply:
I'm trying to avoid logic specific to a given election type in that method. My goal here is to have one pass over the block txs where I check for all concluded elections, and then let `end_block` do what it needs to with those results. If I have `approved_elections` return a validator set for an approved `ValidatorElection`, then I need to add custom logic to handle that election class differently than the others.